### PR TITLE
Optimized named exports

### LINKED ISSUE #134: Component/index.js commited
Hey, I don't know why but, when I build the project all the index.js files are different from those in the repo, even if I didn't touch the matching jsx file.

The two main differences are : 
1. The _extends function which is a generated as a one liner on my machine.
2. The call to _extends which is not indented the same way.

As these files are in the git repo I can't just commit my changes without taking extra care of not committing the compiled version of the jsxs.

Do we really need to publish them to git ?

### PR DESCRIPTION
**Pull request by cromefire**
Fixed #134

### CODE REVIEW

File: ts/IconButton/index.tsx [hunk 1]
```diff
@@ -4,6 +4,8 @@ import {h} from 'preact';
 import MaterialComponent from '../Base/MaterialComponent';
 import Icon from '../Icon';
 
+export {Icon as IconButtonIcon} from '../Icon';
```
Comment (YePpHa) [inline ts/IconButton/index.tsx]: What's the purpose of this? Does it really make sense to export it from IconButton as well as Icon?

Comment (cromefire) [reply ts/IconButton/index.tsx]: It's not totally needed but it's nice to have and doesn't add any overhead

**Comment by cromefire** [comment]
@prateekbh you can merge this

**Comment by cromefire** [comment]
> Let's not advertise this till 2.0 though

Agree

File: ts/Dialog/index.tsx [hunk 2]
```diff
@@ -2,7 +2,7 @@ import {MDCDialog} from '@material/dialog/';
 import {bind} from 'bind-decorator';
 import {h} from 'preact';
 import MaterialComponent from '../Base/MaterialComponent';
-import Button from '../Button';
+import {Button} from '../Button';
```
Comment (prateekbh) [inline ts/Dialog/index.tsx]: Shouldn't button be a default export?

Comment (cromefire) [reply ts/Dialog/index.tsx]: It has a default export but I don't know, if things like webpack can treeshake when you use the default export

**Pull request merged**